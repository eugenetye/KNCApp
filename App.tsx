import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { NavigationContainer } from '@react-navigation/native';
import {Main} from './src/Main';
import React from 'react';
import { Image } from 'react-native';

import Info_Template from './src/Info_Template';

const Stack = createNativeStackNavigator();

function LogoTitle() {
  return (
      <Image 
        style={{ width: 300, height: 30 , resizeMode: 'contain', marginLeft: 140}}
        source={require('./assets/logo.png')}
      />
  );
}

export default function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator 
        screenOptions={{
          headerStyle: {
            backgroundColor: '#C5DFC5',
          },
          headerTintColor: 'black',
          }}>
        <Stack.Screen name="Kalamazoo Nature Center" component={Main} options={{headerTitle: () => <LogoTitle /> }}/>
        <Stack.Screen

          name="Info_Template"
          component={Info_Template}

          options={{
          headerShown: false,
        }}
      />
      </Stack.Navigator>
    </NavigationContainer>
  );
}
