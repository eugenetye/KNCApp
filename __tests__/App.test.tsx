import React from 'react';
import {create} from 'react-test-renderer';

import App from '../App';

describe('<App />', () => {
  // Test Case ID = 1
  it('correctly renders', async () => {
    const tree = create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Test Case ID = 2
  it('has 1 child', async () => {
    const tree = create(<App />).toJSON();
    if (tree && 'children' in tree) {
      expect(tree.children!.length).toBe(1);
    } else {
      expect("no field children in App JSON").toEqual(null);
    }
  });
});
