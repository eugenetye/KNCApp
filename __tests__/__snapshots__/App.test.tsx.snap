// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> correctly renders 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="#C5DFC5"
        color="black"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Kalamazoo Nature Center"
        titleColor="black"
        topInsetEnabled={false}
        translucent={false}
      >
        <RNSScreenStackHeaderSubview
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          type="center"
        >
          <Image
            source={1}
            style={
              {
                "height": 30,
                "marginLeft": 140,
                "resizeMode": "contain",
                "width": 300,
              }
            }
          />
        </RNSScreenStackHeaderSubview>
      </RNSScreenStackHeaderConfig>
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <RNSScreenNavigationContainer
              style={
                {
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <RNSScreen
                activityState={2}
                collapsable={false}
                gestureResponseDistance={
                  {
                    "bottom": -1,
                    "end": -1,
                    "start": -1,
                    "top": -1,
                  }
                }
                sheetAllowedDetents="large"
                sheetCornerRadius={-1}
                sheetExpandsWhenScrolledToEdge={true}
                sheetGrabberVisible={false}
                sheetLargestUndimmedDetent="all"
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "zIndex": 0,
                  }
                }
              >
                <View
                  accessibilityElementsHidden={false}
                  importantForAccessibility="auto"
                  style={
                    [
                      {
                        "backgroundColor": "rgb(242, 242, 242)",
                        "flex": 1,
                      },
                      [
                        {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </RNSScreen>
            </RNSScreenNavigationContainer>
            <View
              collapsable={false}
              onLayout={[Function]}
              pointerEvents="auto"
              style={
                {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderTopColor": "rgb(216, 216, 216)",
                  "borderTopWidth": 0.5,
                  "bottom": 0,
                  "elevation": 8,
                  "height": 49,
                  "left": 0,
                  "paddingBottom": 0,
                  "paddingHorizontal": 0,
                  "position": null,
                  "right": 0,
                  "transform": [
                    {
                      "translateY": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
              <View
                accessibilityRole="tablist"
                style={
                  {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessibilityLabel="Discover, tab, 1 of 5"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": true,
                    }
                  }
                  accessibilityStates={
                    [
                      "selected",
                    ]
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "flexDirection": "column",
                        "justifyContent": "flex-end",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 1,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 0,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                  </View>
                  <Text
                    numberOfLines={1}
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "textAlign": "center",
                        },
                        {
                          "color": "green",
                        },
                        {
                          "fontSize": 10,
                        },
                        undefined,
                      ]
                    }
                  >
                    Discover
                  </Text>
                </View>
                <View
                  accessibilityLabel="Trails, tab, 2 of 5"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": false,
                    }
                  }
                  accessibilityStates={[]}
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "flexDirection": "column",
                        "justifyContent": "flex-end",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 0,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 1,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                  </View>
                  <Text
                    numberOfLines={1}
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "textAlign": "center",
                        },
                        {
                          "color": "gray",
                        },
                        {
                          "fontSize": 10,
                        },
                        undefined,
                      ]
                    }
                  >
                    Trails
                  </Text>
                </View>
                <View
                  accessibilityLabel="QRcode, tab, 3 of 5"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": false,
                    }
                  }
                  accessibilityStates={[]}
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "flexDirection": "column",
                        "justifyContent": "flex-end",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 0,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 1,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                  </View>
                  <Text
                    numberOfLines={1}
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "textAlign": "center",
                        },
                        {
                          "color": "gray",
                        },
                        {
                          "fontSize": 10,
                        },
                        undefined,
                      ]
                    }
                  >
                    QRcode
                  </Text>
                </View>
                <View
                  accessibilityLabel="Current, tab, 4 of 5"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": false,
                    }
                  }
                  accessibilityStates={[]}
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "flexDirection": "column",
                        "justifyContent": "flex-end",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 0,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 1,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                  </View>
                  <Text
                    numberOfLines={1}
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "textAlign": "center",
                        },
                        {
                          "color": "gray",
                        },
                        {
                          "fontSize": 10,
                        },
                        undefined,
                      ]
                    }
                  >
                    Current
                  </Text>
                </View>
                <View
                  accessibilityLabel="Past, tab, 5 of 5"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": false,
                    }
                  }
                  accessibilityStates={[]}
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "flexDirection": "column",
                        "justifyContent": "flex-end",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 0,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 1,
                          },
                        ]
                      }
                    >
                      <Text />
                    </View>
                  </View>
                  <Text
                    numberOfLines={1}
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "textAlign": "center",
                        },
                        {
                          "color": "gray",
                        },
                        {
                          "fontSize": 10,
                        },
                        undefined,
                      ]
                    }
                  >
                    Past
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
